### ISSUE TITLE
Cannot git clone https://code.x2go.org/git/x2goclient-contrib.git on windows through e2guardian

### ISSUE THREAD

**Issue by opoplawski** [e2guardian/e2guardian#787]
This is with latest v5.5 branch.  When trying to git clone https://code.x2go.org/git/x2goclient-contrib.git on windows I'm getting the following error:
```
fatal: the remote end hung up unexpectedly
error: RPC failed; curl 8 Header without colon
```
I think this is related to E2G's download manager keepalive.  With GIT_TRACE_CURL=True I see:
```
12:28:21.737983 http.c:833              => Send SSL data: .
12:28:21.737983 http.c:792              => Send header, 0000000278 bytes (0x00000116)
12:28:21.737983 http.c:804              => Send header: POST /git/x2goclient-contrib.git/git-upload-pack HTTP/1.1
12:28:21.737983 http.c:804              => Send header: Host: code.x2go.org
12:28:21.737983 http.c:804              => Send header: User-Agent: git/2.42.0.windows.1
12:28:21.737983 http.c:804              => Send header: Accept-Encoding: deflate, gzip, br, zstd
12:28:21.737983 http.c:804              => Send header: Content-Type: application/x-git-upload-pack-request
12:28:21.737983 http.c:804              => Send header: Accept: application/x-git-upload-pack-result
12:28:21.737983 http.c:804              => Send header: Content-Length: 175
12:28:21.737983 http.c:804              => Send header:
12:28:21.737983 http.c:818              => Send data, 0000000175 bytes (0x000000af)
12:28:21.737983 http.c:833              => Send data: 00a2want 9b6c055bd04dce91a49e8b45865347089872c19e multi_ack_
12:28:21.737983 http.c:833              => Send data: detailed no-done side-band-64k thin-pack ofs-delta deepen-si
12:28:21.737983 http.c:833              => Send data: nce deepen-not agent=git/2.42.0.windows.1.00000009done.
12:28:21.737983 http.c:818              <= Recv SSL data, 0000000005 bytes (0x00000005)
12:28:21.737983 http.c:833              <= Recv SSL data: .....
12:28:21.737983 http.c:818              <= Recv SSL data, 0000000001 bytes (0x00000001)
12:28:21.737983 http.c:833              <= Recv SSL data: .
12:28:21.737983 http.c:845              == Info: TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
12:28:21.737983 http.c:818              <= Recv SSL data, 0000000217 bytes (0x000000d9)
12:28:21.737983 http.c:833              <= Recv SSL data: ...........4...........C.?....q|..x.............#..*h.B_.BA.
12:28:21.737983 http.c:833              <= Recv SSL data: .Q..1....%......BZ..,cTZ&pD.e.\..b...)........G..4.L..}.Ik.A
12:28:21.737983 http.c:833              <= Recv SSL data: .P...L..8h/...}:h......X..P.....5..........58..l.K7..4&.3...
12:28:21.737983 http.c:833              <= Recv SSL data: ....<...7.5M..}.xO}......Y>. >....*..
12:28:21.737983 http.c:845              == Info: old SSL session ID is stale, removing
12:28:21.737983 http.c:818              <= Recv SSL data, 0000000005 bytes (0x00000005)
12:28:21.737983 http.c:833              <= Recv SSL data: .....
12:28:21.737983 http.c:818              <= Recv SSL data, 0000000001 bytes (0x00000001)
12:28:21.737983 http.c:833              <= Recv SSL data: .
12:28:21.737983 http.c:845              == Info: TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
12:28:21.737983 http.c:818              <= Recv SSL data, 0000000217 bytes (0x000000d9)
12:28:21.737983 http.c:833              <= Recv SSL data: ........D.B\...........C.?....q|..x.....9.Q...c.T..h...Jt..~
12:28:21.737983 http.c:833              <= Recv SSL data: ~,...1".q\.S.nzz.mT..p.....6.......K%.........|..O.|.U.U..m.
12:28:21.737983 http.c:833              <= Recv SSL data: `................!l.|3.Z.,.w....H.......9..@.........PA*.'x]
12:28:21.737983 http.c:833              <= Recv SSL data: .....5.Pd..D:6H...).f..emC"...mVn[h..
12:28:21.737983 http.c:845              == Info: old SSL session ID is stale, removing
12:28:43.864144 http.c:818              <= Recv SSL data, 0000000005 bytes (0x00000005)
12:28:43.864144 http.c:833              <= Recv SSL data: ...."
12:28:43.879450 http.c:818              <= Recv SSL data, 0000000001 bytes (0x00000001)
12:28:43.879450 http.c:833              <= Recv SSL data: .
12:28:43.879450 http.c:792              <= Recv header, 0000000017 bytes (0x00000011)
12:28:43.879450 http.c:804              <= Recv header: HTTP/1.1 200 OK
12:28:43.879450 http.c:818              <= Recv SSL data, 0000000005 bytes (0x00000005)
12:28:43.879450 http.c:833              <= Recv SSL data: ....%
12:28:43.879450 http.c:818              <= Recv SSL data, 0000000001 bytes (0x00000001)
12:28:43.879450 http.c:833              <= Recv SSL data: .
12:28:43.879450 http.c:792              <= Recv header, 0000000020 bytes (0x00000014)
12:28:43.879450 http.c:804              <= Recv header: X-E2GKeepAlive: on
12:28:46.250278 http.c:818              <= Recv SSL data, 0000000005 bytes (0x00000005)
12:28:46.250278 http.c:833              <= Recv SSL data: ...."
12:28:46.250278 http.c:818              <= Recv SSL data, 0000000001 bytes (0x00000001)
12:28:46.250278 http.c:833              <= Recv SSL data: .
12:28:46.250278 http.c:845              == Info: Header without colon
12:28:46.250278 http.c:845              == Info: Closing connection
12:28:46.250278 http.c:818              <= Recv SSL data, 0000000005 bytes (0x00000005)
12:28:46.250278 http.c:833              <= Recv SSL data: ....*
12:28:46.250278 http.c:818              <= Recv SSL data, 0000000001 bytes (0x00000001)
12:28:46.250278 http.c:833              <= Recv SSL data: .
12:28:46.250278 http.c:818              => Send SSL data, 0000000005 bytes (0x00000005)
12:28:46.250278 http.c:833              => Send SSL data: .....
12:28:46.250278 http.c:818              => Send SSL data, 0000000001 bytes (0x00000001)
12:28:46.250278 http.c:833              => Send SSL data: .
12:28:46.250278 http.c:845              == Info: TLSv1.3 (OUT), TLS alert, close notify (256):
12:28:46.250278 http.c:818              => Send SSL data, 0000000002 bytes (0x00000002)
12:28:46.250278 http.c:833              => Send SSL data: ..
error: RPC failed; curl 8 Header without colon
fatal: the remote end hung up unexpectedly
```
If I bypass E2G and use squid directly it works.

**Comment by philipianpearce** [comment]
Thanks for reporting this.   It appears that the error is caused  by an incomplete header being output, so that client is rejecting first data line as 'wrong header line'.    Have raised and fixed bug #788 in v5.5.

Have tested the git clone and now works ok (on MacOS client).  So should also work with windows client.

**Comment by opoplawski** [comment]
Confirmed that it fixed it for me as well.  Thank you!
